src/pages/Signup/UserInfo.js

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+
+const UserInfo = () => {
+  const navigate = useNavigate();
+
+  const userData = () => {
+    fetch(`http://10.58.5.43:8000/users/additional-info`, {
+      method: 'PATCH',
+      headers: { Authorization: localStorage.getItem('Token') },
+      body: JSON.stringify({
+        last_name: LastName,
+        first_name: FirstName,
+        phone_number: PhoneNumber,
+        birth_day: BirthDay,
+      }),
+    })
+      .then(response => response.json())
+      .then(data => {
+        if (data.message === 'USER_INFO_UPDATED') {
+          alert('회원가입에 성공했습니다!');
+          navigate('/main');
+        } else if (data.message === 'KEY_ERROR') {
+          alert(MESSAAGE_ALERT.KEY_ERROR);
+        } else if (data.message === 'FISRT_NAME_ERROR') {
+          alert(MESSAAGE_ALERT.FISRT_NAME_ERROR);
+        } else if (data.message === 'LAST_NAME_ERRO') {
+          alert(MESSAAGE_ALERT.LAST_NAME_ERRO);
+        } else if (data.message === 'PHONE_NUMBER_ERROR') {
+          alert(MESSAAGE_ALERT.PHONE_NUMBER_ERROR);
+        } else if (data.message === 'PHONE_NUMBER_ALREADY_EXIST') {
+          alert(MESSAAGE_ALERT.PHONE_NUMVER_ALREADY_EXIST);
+        }
+      });
+  };
+
+  const [inputValue, setInputValue] = useState({
+    LastName: '',
+    FirstName: '',
+    PhoneNumber: '',
+    BirthDay: '',
+  });
+
+  const { LastName, FirstName, PhoneNumber, BirthDay } = inputValue;
+
+  const handleInput = e => {
+    const { name, value } = e.target;
+    setInputValue({
+      ...inputValue,
+      [name]: value,
+    });
+  };
+
+  const regExp = /^[ㄱ-ㅎ|가-힣]+$/;
+  const regExpPhonNumber = /^01([0|1|6|7|8|9])-([0-9]{3,4})-([0-9]{4})$/;
+
+  const isValidInput =
+    regExp.test(LastName) &&
+    regExp.test(FirstName) &&
+    regExpPhonNumber.test(PhoneNumber) &&
+    BirthDay.length !== 0;
+
+  return (
+    <div>
+      <form className="signUpInput">
+        {SignUpData.map(list => {
+          return (
+            <InputContainer
+              key={list.id}
+              id={list.id}
+              name={list.name}
+              type={list.type}
+              placeholder={list.placeholder}
+              autocomplete={list.autocomplete}
+              onChange={handleInput}
+            />
+          );
+        })}
```

Comment:
```suggestion
        {SignUpData.map(({id, name, type, placeholder, autocomplete}) => {
          return (
            <InputContainer
              key={id}
              id={id}
              name={name}
              type={type}
              placeholder={placeholder}
              autocomplete={autocomplete}
              onChange={handleInput}
            />
          );
        })}
```

이렇게 구조분해할당을 사용해 주는 것이 가독성과 유지보수에 좋습니다!

Reply:
수정했습니다:)